src/domain/usecases/GetAllAccesibleModulesUseCase.ts

```diff
@@ -0,0 +1,46 @@
+import { FutureData } from "../../data/api-futures";
+import { AMRSurveyModule } from "../entities/AMRSurveyModule";
+import { ModuleRepository } from "../repositories/ModuleRepository";
+import _ from "../../domain/entities/generic/Collection";
+import { Future } from "../entities/generic/Future";
+import { NamedRef } from "../entities/Ref";
+import { Id } from "@eyeseetea/d2-api";
+
+export class GetAllAccesibleModulesUseCase {
+    constructor(private moduleRepository: ModuleRepository) {}
+
+    public execute(
+        currentUserGroups: NamedRef[],
+        currentOrgUnitId: Id
+    ): FutureData<AMRSurveyModule[]> {
+        return this.moduleRepository.getAll().flatMap(modules => {
+            return this.moduleRepository
+                .getProgramsEnrolledInOrgUnit(currentOrgUnitId)
+                .flatMap(programs => {
+                    const accessibleModules = modules.map(module => {
+                        const accesibleSurveysInModule = module.surveyPrograms.filter(sp =>
+                            programs.some(p => sp.id === p)
+                        );
+                        //Access logic - A module is accessible to a user only if
+                        if (
```

Comment:
I think the if is a bit complicated to read (although the comments are great), perhaps it would be better if they were variables with descriptive names